lib/internal/abort_controller.js

```diff
@@ -243,24 +254,27 @@ class AbortSignal extends EventTarget {
       }
       signal[kDependantSignals] ??= new SafeSet();
       if (!signal[kComposite]) {
-        resultSignal[kSourceSignals].add(new SafeWeakRef(signal));
+        const signalWeakRef = new SafeWeakRef(signal);
+        finalizers.register(resultSignal, signalWeakRef);
+        resultSignal[kSourceSignals].add(signalWeakRef);
         signal[kDependantSignals].add(resultSignalWeakRef);
       } else if (!signal[kSourceSignals]) {
         continue;
       } else {
-        for (const sourceSignal of signal[kSourceSignals]) {
-          const sourceSignalRef = sourceSignal.deref();
-          if (!sourceSignalRef) {
+        for (const sourceSignalWeakRef of signal[kSourceSignals]) {
```

Comment:
Why was this variable renamed?

Reply:
To make it simpler to understand. `signal[kSourceSignals]` holds weak refs, not the actual `AbortSignal` instances.